cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/distributed/SparkExecutionService.java

```diff
@@ -85,7 +85,7 @@
   public SparkExecutionService(LocationFactory locationFactory, String host,
                                ProgramRunId programRunId, @Nullable WorkflowToken workflowToken) {
     this.locationFactory = locationFactory;
-    this.httpServer = NettyHttpService.builder(SparkExecutionService.class.getName())
+    this.httpServer = NettyHttpService.builder(programRunId.getProgram())
```

Comment:
Don't just use the program name. It is the name for the http server, so better be `programRunId.getProgram() + "-spark-exec-service"`. The name is not just for logging, but also used to name threads.